src/test/java/jenkins/bouncycastle/EncodignDecodingTest.java

```diff
@@ -224,5 +226,15 @@ public void testCreationFromObjectPrivateKeyPEM() throws Exception {
 
         assertEquals(FileUtils.readFileToString(PRIVATE_KEY_PEM), FileUtils.readFileToString(pemFileNew));
     }
+    
+    
+    @Test
+    @Issue(value="JENKINS-35661") 
+    public void testReadKeyPairFromPCKS8PEM() throws Exception {
+        PEMEncodable pemEnc = PEMEncodable.read(getResourceFile("private-key-pcks8.pem"));
+        assertNotNull(pemEnc.toKeyPair());
+        assertNotNull(pemEnc.toPrivateKey());
+        assertNotNull(pemEnc.toPublicKey());
```

Comment:
@alvarolobato 

I had a case where I was doing something similar... look in [this](https://github.com/cloudbees/operations-center-server/pull/102/files) for `// TODO figure out a less hacky way to recover the public key`